### PR TITLE
Implement test functionality for single test files

### LINKED ISSUE #20885: Implement running tests for single bal files
**Description:**
Using the test command on single bal files needs to be implemented

**Steps to reproduce:**
`$ballerina run <test.bal>` 

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by Ibaqu**
## Purpose
The functionality to test single test files needed to be implemented. 

Fixes #20885 

## Approach
Removed restrictions on running single test files. The implementation for testing single files exists partially in the codebase. The functionality was re-implemented with some small changes.

With this implementation, a test file can be individually run without it having to be inside a Tests folder in a ballerina project.

This implementation is aimed at running tests outside of a Ballerina project. Running of individual test files within the test folder of a Ballerina project is restricted. 

## Samples
Updated the test report to display only the tested file. 

```
ballerina test main_test.bal
Compiling source
        main_test.bal
[BTestRunner] [buildSuites] Running build suites

Running tests
        main_test.bal

        [fail] testAssertStringEquals:
            string values not equal: expected 'JohnDoe' but found 'JaneDoe' 
                at ballerina.test:createBallerinaError(assert.bal:41)
                ballerina.test:assertEquals(assert.bal:82)
                main_test:testAssertStringEquals(main_test.bal:29)

        [fail] testAssertFloatEquals:
            float values not equal: expected '30.05' but found '30.0' 
                at ballerina.test:createBallerinaError(assert.bal:41)
                ballerina.test:assertEquals(assert.bal:82)
                main_test:testAssertFloatEquals(main_test.bal:20)

        [pass] testAssertFail
        [pass] testAssertFalse
        [pass] testAssertTrue
        [pass] testAssertNotEqualsJson
        [pass] testAssertNotEqualsString
        [pass] testAssertFloatArrayEquals
        [pass] testAssertIntArrayEquals
        [pass] testAssertStringArrayEquals
        [pass] testAssertBooleanEquals
        [pass] testAssertJsonEquals
        [pass] testAssertIntEquals

        11 passing
        2 failing
        0 skipped
```

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/BTestRunner.java [hunk 1]
```diff
@@ -331,9 +336,9 @@ public void packageProcessed(BLangPackage bLangPackage, TestarinaClassLoader cla
         // TODO the below line is required since this method is currently getting explicitly called from BTestRunner
         TestSuite suite = TesterinaRegistry.getInstance().getTestSuites().get(bLangPackage.packageID.toString());
 
-        if (!bLangPackage.hasTestablePackage()) {
-            return;
-        }
+//        if (!bLangPackage.hasTestablePackage()) {
+//            return;
+//        }
```
Comment (hevayo) [inline misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/BTestRunner.java]: If the code is not required shall we remove ? 

What will be the behavior if you run `ballerina test moduleName` if the module do not have tests.

Comment (Ibaqu) [reply misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/BTestRunner.java]: I will remove this particular block out. 

When we run `ballerina test <moduleName>`, we get the following output on the command line
```
Compiling source
        mohamedaquibzulfikar/module1:0.1.0

Creating balos
        target/balo/module1-2019r3-any-0.1.0.balo

Running tests
        mohamedaquibzulfikar/module1:0.1.0
        No tests found
```

Comment (Ibaqu) [reply misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/BTestRunner.java]: Done

**Comment by Ibaqu** [comment]
The Ballerina test output is modified to display more data on the console. The list of tests that have passed are shown and also the tested file name is displayed for single tests. 

Fixes : https://github.com/ballerina-platform/ballerina-lang/issues/18560

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/TestCommand.java [hunk 2]
```diff
@@ -181,13 +183,57 @@ public void execute() {
...
-                    false);
-            CommandUtil.exitError(this.exitWhenFinish);
+            // when a single bal file is provided
+            // Check if path given is an absolute path. Update the root accordingly
+            if (Paths.get(this.argList.get(0)).isAbsolute()) {
+                sourcePath = Paths.get(this.argList.get(0));
+            } else {
+                sourcePath = sourceRootPath.resolve(this.argList.get(0));
+            }
+
```
Comment (Dilhasha) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/TestCommand.java]: @Ibaqu Shall we get rid of the unnecessary new lines that are introduced?

**Comment by Dilhasha** [comment]
@Ibaqu Shall we include this functionality to [ballerina-test.help](https://github.com/ballerina-platform/ballerina-lang/blob/master/cli/ballerina-tool/src/main/resources/cli-help/ballerina-test.help) file as well?

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/TestCommand.java (hunk 2 above)
Comment (Ibaqu) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/TestCommand.java]: Done

**Comment by Ibaqu** [comment]
> @Ibaqu Shall we include this functionality to [ballerina-test.help](https://github.com/ballerina-platform/ballerina-lang/blob/master/cli/ballerina-tool/src/main/resources/cli-help/ballerina-test.help) file as well?

Added instructions in the help file for the test command.

**Pull request merged**